packages/flutter/lib/src/services/raw_keyboard.dart

```diff
@@ -512,10 +512,18 @@ class RawKeyboard {
       return;
     }
     if (event is RawKeyDownEvent) {
-      _keysPressed.add(event.logicalKey);
+      _keysPressed[event.physicalKey] = event.logicalKey;
     }
     if (event is RawKeyUpEvent) {
-      _keysPressed.remove(event.logicalKey);
+      // Since the value of the logical key is affected by the modifiers that
```

Comment:
I wasn't really contrasting it, just pointing out why the physical key was being used instead of the logical key.  I agree, though, that the comment is probably too "stream of consciousness".  I've edited it for clarity.